setup.sh

```diff
@@ -10,8 +10,20 @@ if [ $EUID -ne 0 ]; then
   exit 1
 fi
 
-if git submodule status --recursive | grep -q "^-"; then
-  git submodule update --init --recursive
+tmpfile=$(mktemp)
+# any error messages from this command will stand out
+# more clearly when run as a separate command rather than
+# being piped
+git submodule status --recursive > "$tmpfile"
+
+if grep -q "^-" "$tmpfile"; then
+  sudo -u $SUDO_USER git submodule update --init --recursive
```

Comment:
@jeffng-or @maliberty Fascinating that `sudo -u $SUDO_USER` prefix was missing up until now...

This causes all sorts of mystical errors that users must have silently suffered through fixed by fixing permissions....

It is not normally visible for experts who have had ORFS installed since forever, but occasional/first time hackers of ORFS must have gotten snagged on this...

Reply:
Why does updating ORFS need sudo?

Reply:
It is the oposite, it needs NOT to use sudo.

Recall that  `sudo ./setup.sh` is run, `sudo -u $SUDO_USER ...` drops out of sudo and runs it as the user.